db/fixtures/ae_datastore/ManageIQ/Cloud/Orchestration/Operations/Methods.class/__methods__/available_vdc_networks.rb

```diff
@@ -0,0 +1,47 @@
+class AvailableVdcNetworks
+  def initialize(handle = $evm)
+    @handle = handle
+  end
+
+  def main
+    fill_fialog_field(fetch_list_data)
+  end
+
+  def fetch_list_data
+    service = @handle.root.attributes["service_template"] || @handle.root.attributes["service"]
+    vdc_networks = service.try(:orchestration_manager).try(:cloud_networks).try(:select) do |net|
+      net.type == "ManageIQ::Providers::Vmware::NetworkManager::CloudNetwork::OrgVdcNet"
+    end
+
+    return nil => "<none>" if vdc_networks.blank?
+
+    vdc_networks_list = { nil => "<select>" }
+    vdc_networks.each { |net| vdc_networks_list[net.ems_ref] = net.name } if vdc_networks
```

Comment:
You were correct; I have added additional test where only vapp network is present in the provider and it works properly in this case as well.